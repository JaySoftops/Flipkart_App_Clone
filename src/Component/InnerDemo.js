import React, { Component } from 'react'

export default class InnerDemo extends Component {
    render() {

        const classes = {
            table: {
                border: "1px solid black",
                margin: "10px"
            },
            tr: {
                textAlign: "center",
            },
            th: {
                width: "100px",
                border: "1px solid black",
                padding: "10px"
            },
            td: {
                width: "100px",
                border: "1px solid black",
                padding: "10px"
            },
        }

        return (
            <table className="custom-border" style={classes.table}>
                <tr style={classes.tr}>
                    <th style={classes.th}>S.No</th>
                    <th style={classes.th}>Name</th>
                    <th style={classes.th}>Email</th>
                </tr>
                <tr style={classes.tr}>
                    <td style={classes.td}>{this.props.State}</td>
                    <td style={classes.td}>Jai</td>
                    <td style={classes.td}>Jai@12344</td>
                </tr>
            </table>
        )
    }
}
