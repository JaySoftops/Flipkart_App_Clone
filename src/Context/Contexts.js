import React, { Component, createContext } from 'react'

const MyContext = createContext(null)

class Contexts extends Component {
    constructor(props) {
        super(props)
        this.state = {
            getData: [],
            isLoading: true,
            FName: 'Jai',
            // Age: 24,
            // Id: 'S78',
            // Tech: 'React',
            singleProduct: null,
            Arr: [1, 2, 3, 4, 5]
        }
    }

    GetApi = async () => {
        try {
            const response = await fetch('https://fakestoreapi.com/products', {
                // const response = await fetch('https://api.escuelajs.co/api/v1/products', {
                method: 'GET'
            })
            const result = await response.json()
            this.setState({ getData: result }, () => {
                console.log(this.state.getData, "@@@@@")
            })
            this.setState({ isLoading: false })
            // console.log(result)
        }
        catch (error) {
            console.log(error)
        }
    }

    componentDidMount() {
        this.GetApi()
    }

    // componentDidUpdate() {
    //     this.GetApi()
    // }

    render() {
        // console.log(this.state.number)
        return (
            <>
                <MyContext.Provider value={this.state}>
                    {this.props.children}
                </MyContext.Provider>
            </>
        )
    }
}

export { Contexts, MyContext }