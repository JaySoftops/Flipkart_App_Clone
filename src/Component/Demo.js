import React, { Component } from 'react'
import DemoController from './DemoController'
import InnerDemo from './InnerDemo';

class Demo extends DemoController {
  render() {
    const classes = {
      demoBody: {
        width: "500px",
        height: "280px",
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        margin: "auto",
        padding: "20px"
      },
      demo: {
        fontSize: "2rem",
        margin: "15px"
      },

      demoButton: {
        width: "130px",
        padding: "10px 0px",
        border: "none",
        borderRadius: "5px",
        boxShadow: "rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.23) 0px 3px 6px",
      }
    }

    return (
      <div className="demoBody" style={classes.demoBody} >
        <div className="demo" style={classes.demo}>Demo Body
          <span className="span" style={classes.span}>span</span>
        </div>
        <button className={`demoButton-${this.state.count}`} style={classes.demoButton} onClick={this.handleAdd}>Increment {this.state.count}</button>
        <div>
          <h3 style={{ textAlign: "center", padding: "15px 0px 0px" }}>Child Component</h3>
          <input type="text" value="Data" style={{ width: "150px", marginLeft: "100px" }} />
          <InnerDemo Sno={this.state.count} />
        </div>
      </div >
    )
  }
}

export default (Demo);