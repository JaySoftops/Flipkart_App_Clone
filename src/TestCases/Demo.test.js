import Demo from "../Component/Demo";
import { mount, render, shallow } from "enzyme";
import { Button } from "@material-ui/core";
import InnerDemo from "../Component/InnerDemo";
import * as math from "../Context/Moc";

describe.skip("Demo Page Test", () => {                                                                                    // describe the wrapper function of whole file's test cases.

    test("demo file's content check", () => {
        let wrapper = shallow(<Demo />)                                                                               // shallow method renders all the child component, it lets us render elements only one level deep "first level"
        // expect(wrapper.contains(<div className="demo">Demo Body</div>)).toEqual(true);
        console.log('Hi everyone');
        // console.log(setProps());
        // wrapper.setProps({name: "Jai"});
        console.log(wrapper.debug());
    });

    // test("Full DOM rendering", () => {
    //     // sinon.spy(Foo.prototype,'componentDidMount');

    //     const wrapper = mount(<Demo />)                                                                           // mount method renders the Full DOM, including Parent and Child Component, it test whole life cycle method of react app.
    //     expect(Foo.prototype.componentDidMount.calledOnce).toEqual(true);                               
    //     console.log(wrapper.debug());
    // });

    test("Static rendering by Render method", () => {
        let wrapper = render(<Demo />)
        console.log('Render method');
        console.log(wrapper.html());
    });

    test("Dive and simulate", () => {
        const wrapper = shallow(<Demo />);
        expect(wrapper.find(".span").length).toBe(1);
        // expect(wrapper.find(Button).dive().find('.MuiButton-text').length).toBe(1);
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');                                                                      // Show 3 time's result by using number of time used this method
    });

    // "it" or "test" is a method to write the test cases and it has 2 arguments,
    // first one is for description 
    // second one is for a callback function that contains the actual test logic. 
    it("click count", () => {
        let wrapper = shallow(<Demo />);
        // find method find the class name and confirm the number of time classname by toEqual
        expect(wrapper.find('.demoButton-0').length).toEqual(1);

        // simulate method used for onClick event
        wrapper.find('.demoButton-0').simulate('click');

        // this statement has updated class number because of simulate method, without simulate this test case is failed.
        expect(wrapper.find('.demoButton-1').length).toEqual(1);
    });

    it("dive, check child component class name", () => {
        let wrapper = shallow(<Demo />)

        // dive method is used for child component element or classname or other things.
        expect(wrapper.find(InnerDemo).dive().find('.custom-border').length).toBe(1);
    });

    it("snapshot 1", () => {
        let wrapper = shallow(<InnerDemo />)

        // snapshot method creates or take a screenshot of file in the snap file,snapshot folder.
        // expect(wrapper).toMatchSnapshot();
    });

    it("snapshot 2", () => {
        let wrapper = shallow(<InnerDemo />)
        // expect(wrapper).toMatchSnapshot();
    });

    it("snapshot 3", () => {
        let wrapper = shallow(<InnerDemo />)
        // expect(wrapper).toMatchSnapshot();
    });

    it("snapshot of Demo file", () => {
        let wrapper = shallow(<Demo />)
        // expect(wrapper).toMatchSnapshot();
    });
});


describe.skip("Another describe test cases of Demo file (shallow vs mount)", () => {

    // it("snapshot 1", () => {
    //     let wrapper = mount(<Demo/>)
    //     // expect(wrapper).toMatchSnapshot();
    //     console.log(wrapper.debug());
    // });

    it("instance testing", async () => {
        let wrapper = shallow(<Demo />);

        // instance method returns the single-node wrapper's node's underlying class instance. It must be a single-ndoe wrapper.
        let anyFn = wrapper.instance().handleObject();
        console.log(anyFn);
        expect(anyFn).toEqual({ id: 4 });
    });

    it("load data function, take time for some output", async () => {
        let wrapper = shallow(<Demo />);
        return wrapper.instance().handleLoadData().then(data => {
            console.log(data)
        })
    });

    it("jest spyon componentdidmount", async () => {
        jest.spyOn(Demo.prototype, 'componentDidMount');
        shallow(<Demo />);

        expect(Demo.prototype.componentDidMount).toHaveBeenCalled();
        expect(Demo.prototype.componentDidMount).toHaveBeenCalledTimes(1);
    });

    // it("jest spyon function", async () => {
    //     let wrapper = shallow(<Demo />);
    //     let instance = wrapper.instance();
    //     jest.spyOn(instance,'handleAdd');
    //     wrapper.find('button').simulate('click');

    //     expect(instance.handleAdd).toHaveBeenCalled();
    //     expect(instance.handleAdd).toHaveBeenCalledWith(5,10);
    // });

    it("state check function", () => {
        let wrapper = shallow(<Demo />);
        console.log(wrapper.state());
    });

    it("props check function", () => {
        let wrapper = shallow(<Demo />);
        let inputProp = wrapper.find('input').props();
        console.log(inputProp);
        let matchProps = {
            type: 'text',
            value: 'Data'
        }
        expect(inputProp).toEqual(matchProps);
    })
});

// Use jest.mock() to mock the module containing the functions
jest.mock("../Context/Moc", () => {
    return {
        add: jest.fn(),
        subtract: jest.fn(),
        multiply: jest.fn()
    }
});


describe(" For Mock Function", () => {

    beforeAll(() => {  
        console.log('1, I am executing BEFORE ALL test cases')
    });
    afterAll(() => {
        console.log('2, I am executing AFTER ALL test cases')
    });
    beforeEach(() => {
        console.log('3, I am executing BEFORE EACH test cases')
    });
    afterEach(() => {
        console.log('4, I am executing AFTER EACH test cases')
    });

    // it(" API Data test", () => {                                        // Not Working
    //     const mockAdd = jest.fn();
    //     mockAdd.mockReturnValue(5);
    //     const result = math.add(3,2);
    //     expect(mockAdd).toHaveBeenCalledWith(3,2);
    //     expect(result).toEqual(5);
    //     // console.log('Mockss'); 
    // });

    it(" Adding ", () => {

        // Provide custom implementation for the add function
        math.add.mockImplementation(() => 40);
        console.log(math.add(9, 18));
    });

    it(" Subtract ", () => {

        // Provide custom implementation for the subtract function
        math.subtract.mockImplementation(() => 22);
        console.log(math.subtract(9, 21));
    });

    it(" Multiply ", () => {

        // Provide custom implementation for the multiply function
        math.multiply.mockImplementation(() => 48);
        console.log(math.multiply(2, 44));
    });
});