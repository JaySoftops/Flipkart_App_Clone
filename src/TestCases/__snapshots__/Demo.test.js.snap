// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Another describe test cases of Demo file (shallow vs mount) snapshot 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Another describe test cases of Demo file (shallow vs mount) snapshot 1 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Another describe test cases of Demo file (shallow vs mount) snapshot 2 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Another describe test cases of Demo file (shallow vs mount) snapshot 4 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Demo Page Test snapshot 1 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Demo Page Test snapshot 2 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Demo Page Test snapshot 3 1`] = `
<table
  className="custom-border"
  style={
    Object {
      "border": "1px solid black",
      "margin": "10px",
    }
  }
>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      S.No
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Name
    </th>
    <th
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Email
    </th>
  </tr>
  <tr
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    />
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai
    </td>
    <td
      style={
        Object {
          "border": "1px solid black",
          "padding": "10px",
          "width": "100px",
        }
      }
    >
      Jai@12344
    </td>
  </tr>
</table>
`;

exports[`Demo Page Test snapshot of Demo file 1`] = `
<div
  className="demoBody"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "280px",
      "justifyContent": "center",
      "margin": "auto",
      "padding": "20px",
      "width": "500px",
    }
  }
>
  <div
    className="demo"
    style={
      Object {
        "fontSize": "2rem",
        "margin": "15px",
      }
    }
  >
    Demo Body
    <span
      className="span"
    >
      span
    </span>
  </div>
  <button
    className="demoButton-0"
    onClick={[Function]}
    style={
      Object {
        "border": "none",
        "borderRadius": "5px",
        "boxShadow": "rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.23) 0px 3px 6px",
        "padding": "10px 0px",
        "width": "130px",
      }
    }
  >
    Increment 
    0
  </button>
  <div>
    <h3
      style={
        Object {
          "padding": "15px 0px 0px",
          "textAlign": "center",
        }
      }
    >
      Child Component
    </h3>
    <InnerDemo
      Sno={0}
    />
  </div>
</div>
`;
